### PR TITLE
GO-2030 refactor techSpace/spaceService

### PR DESCRIPTION
**Pull request by cheggaaa**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: space/create.go [hunk 1]
```diff
@@ -19,15 +21,14 @@ func (s *service) create(ctx context.Context, coreSpace *spacecore.AnySpace) (Sp
 	// create mandatory objects
 	_, err := s.provider.DeriveObjectIDs(ctx, coreSpace.Id(), sbTypes)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("DeriveObjectIDs error: %v; spaceId: %v", err, coreSpace.Id())
```
Comment (deff7) [inline space/create.go:24]: maybe wrap error via `%w`?

**Review by mcrakhman — APPROVED** [review]

File: space/load.go [hunk 2]
```diff
@@ -10,65 +11,76 @@ import (
...
+	}
 	s.loading[spaceID] = newLoadingSpace(s.ctx, s.open, spaceID, s.onLoad)
-	loaderCreated = true
 	return
 }
 
 func (s *service) onLoad(spaceID string, sp Space, loadErr error) (err error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
+
```
Comment (deff7) [inline space/load.go:43]: maybe use `errors.Is` here? It will be less fragile if someone accidentally wrap `loadErr`

File: space/create.go (hunk 1 above)
Comment (cheggaaa) [reply space/create.go]: fixed

File: space/load.go (hunk 2 above)
Comment (cheggaaa) [reply space/load.go:44]: Yep, also replaced in other places

**Review by deff7 — APPROVED** [review]

**Pull request merged**